rfqmsg/messages.go

```diff
@@ -90,6 +90,63 @@ func (id *ID) Record() tlv.Record {
 	return tlv.MakeStaticRecord(0, id, recordSize, IdEncoder, IdDecoder)
 }
 
+// TransferType defines the type of transaction which will be performed if the
+// quote request leads to an accepted agreement.
+type TransferType uint8
```

Comment:
why are we adding this? I don't see why we need to describe the purpose of the transfer via this enum

Reply:
This field may not be strictly necessary, in my opinion. However, after discussing with @guggero, I believe we're including it for two reasons:  
1. Debugging: For asset-to-asset quotes (rather than asset-to-BTC) later on, it could provide useful context.  
2. Future use: This field might support adding other transfer types in the future.

Reply:
Correct. But we should also actually set and use it. My goal is to replace the following code with the transfer type detection:

https://github.com/lightninglabs/taproot-assets/blob/c7273399e009a4b6b710c59b818e34c559773130/rfqmsg/request.go#L512-L551